relays/substrate-client/src/finality_source.rs

```diff
@@ -30,26 +31,21 @@ use std::{marker::PhantomData, pin::Pin};
 
 /// Wrapped raw Justification.
 #[derive(Debug, Clone)]
-pub struct Justification<Header> {
+pub struct Justification<Number> {
+	target_header_number: Number,
```

Comment:
This is always going to match the number decoded from `raw_justification`, right?
```suggestion
	/// Header number decoded from the [`raw_justification`].
	target_header_number: Number,
	/// Raw, encoded justification bytes.
```